open-apparel-registry/open-apparel-registry
# Update train model command

## Overview

This issue creates a train_and_activate_gazetteer function. We consolidate all code related to creating and training a model into this function, allowing us to clean up other sections of legacy code.

Connects #2037

## Demo
```
$ ./scripts/manage train_model
Creating open-apparel-registry_django_run ... done
reading training from file
Final predicate set:
SimplePredicate: (firstIntegerPredicate, address)
Final predicate set:
TfidfTextSearchPredicate: (0.6, address)
SimplePredicate: (firstTwoTokensPredicate, name)
SimplePredicate: (oneGramFingerprint, name)
using cross validation to find optimum alpha...
optimum alpha: 0.100000, score 0.5003286163666517
Final predicate set:
SimplePredicate: (sameFiveCharStartPredicate, address)
SimplePredicate: (firstTwoTokensPredicate, name)
SimplePredicate: (twoGramFingerprint, name)
$
```

## Notes

## Testing Instructions

* run `./scripts/resetdb`
* run `./scripts/manage train_model`. The debug statements will cause the script to output the model ID of the previous model, pausing to wait for user input before continuing.:
```
9
Press Enter to continue...
```
* Before we continue with this script, open a new tab and run `./scripts/manage dbshell`
* Run the following 3 insert statements.
 **Please Note:** the table name must be updated to reflect the previous model ID from the other tab.
```sql
insert into dedupe_indexed_records_9 (id, block_key, record_id, record_data) values (99996,'azavea','AZ012346','{"country": "us", "name": "azavea", "address": "123 fake street phila pa 19111"}');
```
```sql
insert into dedupe_indexed_records_9 (id, block_key, record_id, record_data) values (99997,'azavea','AZ012347','{"country": "us", "name": "azavea", "address": "123 fake street phila pa 19111"}');
```
```sql
insert into dedupe_indexed_records_9 (id, block_key, record_id, record_data) values (99999,'azavea','AZ012348','{"country": "us", "name": "azavea", "address": "123 fake street phila pa 19111"}');
```
* Switch back to the `train_model` tab, and press enter to continue the script. the rest of the process will commence, and the script will terminate
* Switching to the database tab, run the following query to verify that the timestamp for the active model is correct:
```sql
select id, created_at,is_active,activated_at from api_trainedmodel;
```
* run the query below to confirm that the rows have been added to the current index table:
```sql
select * from dedupe_indexed_records where record_id = 'AZ012346' OR record_id = 'AZ012347' OR record_id = 'AZ012348';
```
This is an example of what the output should look like:
```sql
  id   |  block_key   | record_id |                                   record_data

-------+--------------+-----------+------------------------------------------------------------------------
----------
 66788 | 123fa:0      | AZ012346  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
 66789 | avazeaveza:2 | AZ012346  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
 66792 | 123fa:0      | AZ012347  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
 66793 | avazeaveza:2 | AZ012347  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
 66796 | 123fa:0      | AZ012348  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
 66797 | avazeaveza:2 | AZ012348  | {"country": "us", "name": "azavea", "address": "123 fake street phila p
a 19111"}
(6 rows)
```

## Checklist
